cmeutils/structure.py

```diff
@@ -8,6 +8,106 @@
+    A_name, B_name, C_name : str
+        Name(s) of particles that form the angle triplet 
+        (found in gsd.hoomd.Snapshot.particles.types)
+        They must be given in the same order as they form the angle
+    start : int
+        Starting frame index for accumulating bond lengths.
+        Negative numbers index from the end. (default 0)
+    stop : int
+        Final frame index for accumulating bond lengths. (default -1)
+
+    Returns
+    -------
+    1-D nump.array 
+        Array of bond angles in degrees
+
+    """
+    angles = []
+    trajectory = gsd.hoomd.open(gsd_file, mode="rb")
+    angle_name = "-".join([A_name, B_name, C_name])
+    for snap in trajectory[start: stop]:
+        if angle_name not in snap.angles.types:
+            raise ValueError(
+                    f"Angle {angle_name} not found in snap.angles.types. "
+                    "A_name, B_name, C_name must match the order "
+                    "as they appear in snap.angles.types."
+                )
+        for idx, angle_id in enumerate(snap.angles.typeid):
+            _angle_name = snap.angles.types[angle_id]
+            if _angle_name == angle_name:
+                pos1 = snap.particles.position[snap.angles.group[idx][0]]
+                img1 = snap.particles.image[snap.angles.group[idx][0]]
+                pos2 = snap.particles.position[snap.angles.group[idx][1]]
+                img2 = snap.particles.image[snap.angles.group[idx][1]]
+                pos3 = snap.particles.position[snap.angles.group[idx][2]]
+                img3 = snap.particles.image[snap.angles.group[idx][2]]
+                pos1_unwrap = pos1 + (img1 * snap.configuration.box[:3])
+                pos2_unwrap = pos2 + (img2 * snap.configuration.box[:3])
+                pos3_unwrap = pos3 + (img3 * snap.configuration.box[:3])
+                u = pos2_unwrap - pos1_unwrap
+                v = pos3_unwrap - pos2_unwrap
+                angles.append(np.round(angle_between_vectors(u, v, False), 3))
+    trajectory.close()
+    return np.array(angles)
+
+
+def bond_distribution(
+    gsd_file, A_name, B_name, start=0, stop=-1
+):
+    """Returns the bond length distribution for a given bond pair 
+    
+    Parameters
+    ----------
+    gsdfile : str
+        Filename of the GSD trajectory.
+    A_name, B_name : str
+        Name(s) of particles that form the bond pair
+        (found in gsd.hoomd.Snapshot.particles.types)
+    start : int
+        Starting frame index for accumulating bond lengths.
+        Negative numbers index from the end. (default 0)
+    stop : int
+        Final frame index for accumulating bond lengths. (default -1)
+
+    Returns
+    -------
+    1-D numpy array
+        Array of bond lengths
+
+    """
+    bonds = []
+    trajectory = gsd.hoomd.open(gsd_file, mode="rb")
+    for snap in trajectory[start:stop]:
+        for idx, bond in enumerate(snap.bonds.typeid):
+            bond_name = snap.bonds.types[bond]
+            if bond_name in [
```

Comment:
Could raise an error if the bond type doesn't exist?

Reply:
good idea, should be added now.